core/src/test/java/org/apache/iceberg/TableTestBase.java

```diff
@@ -562,6 +544,44 @@ void validateDeleteManifest(
     Assert.assertFalse("Should find all files in the manifest", expectedFiles.hasNext());
   }
 
+  private <T extends ContentFile<T>> void validateManifestSequenceNumbers(
```

Comment:
This is a good way to test the core logic. Shall we also add tests at a higher level? For instance, it would be great to check methods for added/removed files in `Snapshot` behave correctly.